stdlib/public/core/StringStorage.swift

```diff
@@ -655,6 +682,7 @@ final internal class __SharedStringStorage
   : __SwiftNativeNSString, _AbstractStringStorage {
   internal var _owner: AnyObject?
   internal var start: UnsafePointer<UInt8>
+  internal let deallocator: String.Deallocator
```

Comment:
Why not have `_owner` be a class whose deinit invokes the closure?

Reply:
I just didn't know what the `_owner` property was used for 🙂   I've taken the whole `Deallocator` out and made it just a closure in the user-facing API.